product_configurator/models/product.py

```diff
@@ -407,21 +413,31 @@ def validate_configuration(self, value_ids, custom_vals=None, final=True):
                 common_vals = set(value_ids) & set(line.value_ids.ids)
                 custom_val = custom_vals.get(attr.id)
                 if line.required and not common_vals and not custom_val:
+                    if do_raise:
+                        raise ValidationError(_("No value provided for %s") % line.attribute.name)
                     # TODO: Verify custom value type to be correct
                     return False
 
         # Check if all all the values passed are not restricted
         for val in value_ids:
             available = self.value_available(
-                val, [v for v in value_ids if v != val])
+                val, [v for v in value_ids if v != val], do_raise=do_raise)
             if not available:
+                # if do_raise is True, we may have already raised a specific error message.
+                # if not, then raise a more generic one...
+                if do_raise:
+                    attr_val = self.env['product.attribute.value'].browse(val)
+                    raise ValidationError(_("%s can not be selected for %s") % (attr_val.name, attr_val.attribute_id.name))
                 return False
 
         # Check if custom values are allowed
         custom_attr_ids = self.attribute_line_ids.filtered(
             'custom').mapped('attribute_id').ids
 
         if not set(custom_vals.keys()) <= set(custom_attr_ids):
```

Comment:
Here maybe we can do invalid_custom_vals = set1 - set2

And if this sets has any values we can pass them in the ValdationError with "Attributes (%s) cannot hold custom values"

What do you think?